django_ai_assistant/decorators.py

```diff
@@ -0,0 +1,35 @@
+import uuid
+from functools import wraps
+
+
+def _cast_id(item_id, model):
+    if isinstance(item_id, str) and "UUID" in model._meta.pk.get_internal_type():
+        return uuid.UUID(item_id)
+    return item_id
+
+
+# Decorator to cast ids to the correct type when using workaround UUIDAutoField
+def with_cast_id(func):
```

Comment:
Non-blocking suggestion: Maybe turn this comment into a docstring?

Example: 

`@admin.register`
https://github.com/django/django/blob/bcc327aa326093a39f01a9bc98198807444900f3/django/contrib/admin/decorators.py#L80-L90

Reply:
Since it's an internal method (it's not gonna be exposed to people using the lib), I think I'll leave it as a comment if that's ok.

Reply:
Sure